### PR TITLE
[Docs] Extend runbook for MimirCompactorHasNotSuccessfullyRunCompaction

### LINKED ISSUE #4608: Docs: Add information on recovering from a scenario where compaction has fallen behind.
#### Is your documentation request related to a feature? If so, which one?

Runbooks -> Compaction

#### Describe the solution that you’d like or the expected outcome

It's possible to get into a state where compaction has fallen way behind and is failing to catch up.

For example: Alerts installed late, and initial cluster setup was misconfigured with compaction failing, but the administrator was unaware.

It would be good to improve the MimirCompactorHasNotSuccessfullyRunCompaction runbook to describe what to do in this scenario.

### PR DESCRIPTION
**Pull request by eamonryan**
#### What this PR does
Improves the MimirCompactorHasNotSuccessfullyRunCompaction runbook

#### Which issue(s) this PR fixes or relates to

Fixes #4608 

Also fixes indentation in the same runbook as it was a little incorrect.

#### Checklist

- [ ] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md
Comment (aknuds1) [inline CHANGELOG.md:44]: This should be the PR number:
```suggestion
* [ENHANCEMENT] Docs: Runbook for MimirCompactorHasNotSuccessfullyRunCompaction extended to include scenario where compaction has fallen behind. #4609
```

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (aknuds1) [inline docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:595]: Are you sure this section should no longer be beneath the _Invalid result block_ bullet point? I'm wondering if it's intentional that this item is nested beneath _Invalid result block_.

**Review by aknuds1 — CHANGES_REQUESTED** [review]
The changelog entry should refer to the PR and not the issue.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [inline docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:611]: This is kind of being repeated in the bullet points below. I think it's easier to grok if the format is the same as the points above

```
- <title>
	- How to detect
	- What it means
	- how to mitigate
```

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:595]: no strong opinion form me, but I'd say that this feels like part of the "invalid result block" bullet point. I think it's clear both ways though.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [inline docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:618]: In this panel the long-term trend is more important than the current number. Maybe suggest to view the dashboard for 7 days and see if the number is steadily increasing.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [inline docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:619]: I'd mention the impact on the store-gateways here

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [inline docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:622]: @colega was considering making this 1 for every 8M AS. I also think 25M is too much.

**Review by dimitarvdimitrov — COMMENTED** [review]
Thanks for the contribution! I have a few comments

**Review by aknuds1 — APPROVED** [review]
LGTM to unblock, but please see Dimitar's comments.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:595]: The multiple layers of bullets (6 layers!) made me think it was a mistake and I found it easier to read in this edited form - but if it's the intention I'm happy to undo the change, whatever people think is clearer :)

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:618]: I think both are valid - you definitely would like to know if it's steadily increasing but in a scenario where you've only just looked at the panel for the first time, if it shows you're at ~50k blocks, well even without a trend that shows you something is wrong, no?

Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:618]: I can change the language to suit both easy enough, though.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:608]: That's a fair point. I was thinking from the perspective of a recent issue seen, but you're right it could happen for any alerts, though this particular one's impact can take longer to recover from, but it's probably clearer to just remove it. As long as people can search the alert and find this section they'll find their answer.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:618]: > A tenant would have one 24h block per day plus smaller blocks to be compacted if all is well.

that's not true for the split&merge compactor. Maybe reword to include that detail too

> Thousands and thousands of blocks is not normal.

Maybe cap this at 50K blocks? or 100K? We have a tenant with 7K and I can image the same tenant cluster with a few years' retention will have a multiple of 7K.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (colega) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:622]: We took 25M because that's what saves the compactor from crashing when compacting, however those huge blocks are still suboptimal for the query path, especially when query sharding is enabled, and lower number is better.

I'd also state that this number should be an even number.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:618]: > that's not true for the split&merge compactor. Maybe reword to include that detail too

How would you ballpark it for split&merge? Afaik once merging has occurred you do still end up with 1 24h block per day, just you may have a larger quantity of smaller blocks that are still undergoing split-and-merge, if that's incorrect let me know :)

> Maybe cap this at 50K blocks? or 100K? We have a tenant with 7K and I can image the same tenant cluster with a few years' retention will have a multiple of 7K.

I debated what number to use here, which is why I tried to add the age of tenant above, if 50k is a safer "silly" number, happy to go with that.

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (eamonryan) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md:622]: Cool, I'll go with 8M then and edit this. Thanks!

File: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md
Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/mimir-runbooks/_index.md]: > How would you ballpark it for split&merge? Afaik once merging has occurred you do still end up with 1 24h block per day, just you may have a larger quantity of smaller blocks that are still undergoing split-and-merge, if that's incorrect let me know :)

you get 1 24hr block for each `-compactor.split-and-merge-shards`. The idea is that each block contains series whose labels hash modulo `-compactor.split-and-merge-shards` is the same.

**Pull request merged**
